packages/react/src/QuestionnaireForm/QuestionnaireForm.tsx

```diff
@@ -641,3 +733,95 @@ function getNumberOfPages(items: QuestionnaireItem[]): number {
   });
   return pages.length > 0 ? items.length : 0;
 }
+
+function repeatableLinkId(linkId: string): string {
+  const repeatablePattern = /-repeatable-(\d+)$/;
+  const match = linkId.match(repeatablePattern);
+
+  if (match) {
+    const currentRepeatableIndex = parseInt(match[1], 10);
+    return linkId.replace(repeatablePattern, `-repeatable-${currentRepeatableIndex + 1}`);
+  } else {
+    return `${linkId}-repeatable-1`;
+  }
+}
+
+function allowRepeatable(item: QuestionnaireItem, answers: Record<string, QuestionnaireResponseItemAnswer>): boolean {
+  if (item.type === QuestionnaireItemType.group) {
+    return true;
+  }
+  const linkId = item.linkId;
+  if (!linkId) {
+    return false;
+  }
+  return !!answers[linkId];
+}
+
+function repeatableInsert(
+  items: QuestionnaireItem[],
+  currentItem: QuestionnaireItem,
+  newItem: QuestionnaireItem
+): QuestionnaireItem[] {
+  for (let i = 0; i < items.length; i++) {
+    if (items[i].linkId === currentItem.linkId) {
+      const updatedItems = [...items.slice(0, i + 1), newItem, ...items.slice(i + 1)];
+      return updatedItems;
+    }
+
+    if (items[i].item) {
+      const updatedNestedItems = repeatableInsert(items[i].item ?? [], currentItem, newItem);
+      if (updatedNestedItems !== items[i].item) {
+        items[i].item = updatedNestedItems;
+      }
+    }
+  }
+  return items;
+}
+
+function removeRecentItem(items: QuestionnaireItem[], currentItem: QuestionnaireItem): QuestionnaireItem[] {
+  for (let i = 0; i < items.length; i++) {
+    if (items[i].linkId === currentItem.linkId) {
+      items.splice(i, 1);
+
+      if (i - 1 >= 0) {
+        items[i - 1].repeats = true;
+      }
+      return items;
+    }
+
+    if (items[i].item) {
+      const updatedNestedItems = removeRecentItem(items[i].item ?? [], currentItem);
+      if (updatedNestedItems !== items[i].item) {
+        items[i].item = updatedNestedItems;
+      }
+    }
+  }
+  return items;
+}
+
+function createRepeatableItem(item: QuestionnaireItem): QuestionnaireItem {
+  let newText = item.text ?? '';
+  if (!newText.endsWith(' continued')) {
+    newText += ' continued';
+  }
+  return {
+    ...item,
+    text: newText,
+    linkId: repeatableLinkId(item.linkId ?? ''),
+    repeats: true,
+    extension: [
+      {
+        url: 'http://hl7.org/fhir/R4/questionnaire-definitions.html#Questionnaire.item.repeats',
+        valueString: 'removable',
+      },
+    ],
+  };
+}
+
+function extensionForRemovingRepeatable(item: QuestionnaireItem): boolean {
+  const extension = getExtension(
+    item,
+    'http://hl7.org/fhir/R4/questionnaire-definitions.html#Questionnaire.item.repeats'
```

Comment:
I needed a way to have the remove show for all of the repeatable components, except the original one, so I created an extension for it. I searched around the fhir specs for something close to it but couldn't find any. Something like "Can remove" or "Original Repeat"

Reply:
Let's discuss this offline.  I'm pretty sure this is not necessary.  I think wires are getting crossed between `QuestionnaireItem` and `QuestionnaireResponseItem`.